client/src/components/NavBar.tsx

```diff
@@ -0,0 +1,208 @@
+import { Fragment, useState } from 'react';
+import { Navigation } from '../lib/types';
+import { Link } from 'react-router-dom';
+import { Dialog, Popover, Transition } from '@headlessui/react';
```

Comment:
Don't you have to pay for headlessui? Oh, I think that one is free, though it doesn't have many components.